mirascope/tools/web/parse_url_content.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+import re
+from typing import ClassVar, TypeVar
+
+import requests
+from bs4 import BeautifulSoup
+from pydantic import Field
+
+from mirascope.tools.base import ConfigurableTool, _ToolConfig
+
+
+class ParseURLConfig(_ToolConfig):
+    """Configuration for URL content parsing"""
+
+    parser: str = Field(
+        "html.parser",
+        description="parser to use for parsing HTML content in BeautifulSoup",
+        examples=["html.parser", "lxml", "lxml-xml", "html5lib"],
+    )
+    timeout: int = Field(
+        default=5,
+        description="Timeout in seconds for URL request",
+    )
+
+    @classmethod
+    def from_env(cls) -> ParseURLConfig:
+        """No environment variables needed for URL parsing"""
+        return cls()  # pyright: ignore [reportCallIssue]
+
+
+_ToolSchemaT = TypeVar("_ToolSchemaT")
+
+
+class ParseURLContent(ConfigurableTool[ParseURLConfig, _ToolSchemaT]):
+    """Tool for parsing and extracting main content from URLs.
+    Fetches content from URL, removes unnecessary elements like scripts, styles, navigation, etc.,
+    and returns clean text content from the webpage's main body.
+    """
+
+    __config__ = ParseURLConfig()  # pyright: ignore [reportCallIssue]
+    __prompt_usage_description__: ClassVar[str] = """
```

Comment:
this reads like usage description for a human reader, but it's supposed to be a usage description for an LLM. Also, we want to make sure that it is succinct and clear (since more tokens cost more money, so we want to do as good a job as possible in as few tokens as possible).